tmt/steps/execute/__init__.py

```diff
@@ -206,6 +206,22 @@ def log(self, test, filename=None, full=False, create=False):
         path = os.path.join(directory, filename)
         return path if full else os.path.relpath(path, self.step.workdir)
 
+    def prepare_tests(self):
+        """
+        Prepare discovered tests for testing
+
+        Check which tests have been discovered, for each test prepare
+        aggregated metadata in a 'test.yaml' file under the test log
+        directory and finally return a list of all discovered tests.
+        """
+        tests = self.step.plan.discover.tests()
+        for test in tests:
+            metadata_filename = self.log(
+                test, filename='metadata.yaml', full=True, create=True)
```

Comment:
Would it make sense to document this or at least note such a file is generated and what is it's purpose?

Reply:
Do you mean just to add more comments to the code or a new section in https://tmt.readthedocs.io/?

Reply:
I'd say this is just purely internal and no external app should realy on the format.

Reply:
The current need is more internal, but it might be documented as official way to provide metadata to tests (test frameworks). It should be stable enough, anyway.

Reply:
Good point. In some way, this really is defining an API between `tmt` and the `framework`. Let's document this under the `execute` step specification.